app-ios/tutanota/Sources/AppDelegate.swift

```diff
@@ -53,51 +49,80 @@ class AppDelegate : UIResponder,
       blobUtils: BlobUtil()
     )
     self.window!.rootViewController = viewController
-    
+
     UNUserNotificationCenter.current().delegate = self
-    
+
     window!.makeKeyAndVisible()
-    
+  }
+
+  func application(
+    _ application: UIApplication,
+    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]?
+  ) -> Bool {
+    TUTSLog("Start Tutanota wit launch options: \(String(describing: launchOptions))")
+    self.main()
     return true
   }
-  
+
   func applicationWillEnterForeground(_ application: UIApplication) {
     UIApplication.shared.applicationIconBadgeNumber = 0
   }
-  
+
   func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
     if let callback = self.pushTokenCallback {
       let stringToken = deviceTokenAsString(deviceToken: deviceToken)
       callback(.success(stringToken!))
       self.pushTokenCallback = nil
     }
   }
-  
+
   func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
     self.pushTokenCallback?(.failure(error))
     self.pushTokenCallback = nil
   }
-  
+
+  /** handles tutanota:// links */
+  func application(_ application: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:] ) -> Bool {
+    switch url.scheme?.appending("://") {
```

Comment:
another reason to build URL properly is to not have `://` in `TUTANOTA_SHARE_SCHEME`.